### DISCUSSION TITLE
CHILD_SA tunnel type difference

### DISCUSSION THREAD

**Discussion by naxoss** [strongswan/strongswan#591] [Q&A] [answered]
Hi,

I wasn't to find response anywhere so I'm gonna give a shot here :) When I list connected IKEv2 on my server using ```swanctl -l```, most CHILD_SAs has listing like ```...  INSTALLED, TUNNEL-in-UDP, ESP:AES_CBC-256/HMAC_SHA1_96...```, but some of them has only ``` ...  INSTALLED, TUNNEL ...```

So I'm wondering what is the difference between ```TUNNEL-in-UDP``` and just ```TUNNEL``` ? Is it something that could affect speed / reliability? 🤷‍♂️ 
(we're using only IKEv2, Strongswan 5.7.2)

Thank you :)

**Comment by tobiasbrunner** [comment, marked as answer]
Please refer to the wiki page about [NAT Traversal](https://wiki.strongswan.org/projects/strongswan/wiki/NatTraversal). UDP encapsulation could also be forced (via config or because the peer's IPsec implementation can only operate on encapsulated packets, like e.g. our Android app).

> Is it something that could affect speed / reliability?

It does add additional overhead (8-byte UDP header per ESP packet). And NATs generally can impact the reliability (they could e.g. get restarted and lose their NAT mappings).

**Comment by Thermi** [comment]
`TUNNEL-in-UDP` uses UDP encapsulation (ESP packet within UDP packet), which is used for NAT traversal.
`TUNNEL` is just ESP.
UDP encapsulation is easier to track for stateful firewalls because both directions use the same ip, port tuple.
For ESP, the relationship between requests and replies is not deducable from the packet information because besides the IPs, the two directions do not share any information (only the IPsec SA is otherwise contained and trackable, but it is unique per direction, so you can't correlate replies to requests).

**Comment by naxoss** [comment]
Thank you guys both for your answers 👍
